images/bridge2/assistant/assistant.go

```diff
@@ -110,13 +227,33 @@ Overall {} is a powerful system that can help humans with a wide range of tasks
 `, "{}", name)
 }
 
-func (a *OpenAIClient) Call(assistant, speaker, input string) (string, error) {
+type OpenAIClient struct {
+	Name          string
+	Endpoint      string
+	SystemMessage string
+}
+
+func OpenAIClientGenerator(endpoint string) func(name, systemMessage string) Client {
+	return func(name, systemMessage string) Client {
+		return OpenAIClient{
+			Name:          name,
+			Endpoint:      endpoint,
+			SystemMessage: systemMessage,
+		}
+	}
+}
+
+func (a OpenAIClient) AssistantName() string {
+	return a.Name
+}
+
+func (a OpenAIClient) Complete(speaker, input string) (string, error) {
 	maxTokens := 4096
 	systemMessage := a.SystemMessage
 	prompt, err := prompts.Render("complete", map[string]any{
```

Comment:
what are your thoughts on switching the "prompt" we store to actually be the template instead?

Reply:
I'm assuming you mean storing the final prompt which was generated, rather than the Go template itself. I'll make that change.
(I suppose we could include the template & its parameters, though then that becomes tied to assuming it's being generated by a Go template, rather than storing the final prompt as text)

Reply:
no, i mean the go template itself. i think we should store the rendered prompt we used as well, but that's for each completion... as opposed what we would store on each "prompt change"